packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -13,6 +18,28 @@ interface SelectTokenButtonProps {
   token?: TokenFromTokenlist;
 }
 
+const HOUR_OPTION = "hour";
+const DAY_OPTION = "day";
+const WEEK_OPTION = "week";
+const MONTH_OPTION = "month";
+
+const frequencyOptions = [
+  { option: HOUR_OPTION, name: "Hour" },
+  { option: DAY_OPTION, name: "Day" },
+  { option: WEEK_OPTION, name: "Week" },
+  { option: MONTH_OPTION, name: "Month" },
+];
+
+const endDateByFrequency: Record<string, number> = {
+  [HOUR_OPTION]: new Date().setDate(new Date().getDate() + 2),
+  [DAY_OPTION]: new Date().setMonth(new Date().getMonth() + 1),
+  [WEEK_OPTION]: new Date().setMonth(new Date().getMonth() + 3),
+  [MONTH_OPTION]: new Date().setFullYear(new Date().getFullYear() + 1),
+};
+const startDateTimeTimestamp = new Date().setMinutes(
+  new Date().getMinutes() + 30
+);
```

Comment:
this logic is okay here for now, but It would be great to have a place for this logic, maybe utils or create a model.